pkg/contexts/ocm/blobhandler/handlers/generic/npmjs/blobhandler.go

```diff
@@ -0,0 +1,127 @@
+package npmjs
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/open-component-model/ocm/pkg/contexts/ocm/accessmethods/npm"
+	"github.com/open-component-model/ocm/pkg/contexts/ocm/cpi"
+	"github.com/open-component-model/ocm/pkg/logging"
+	"github.com/open-component-model/ocm/pkg/mime"
+)
+
+type artifactHandler struct {
+	spec *Config
+}
+
+func NewArtifactHandler(repospec *Config) cpi.BlobHandler {
+	return &artifactHandler{repospec}
+}
+
+func (b *artifactHandler) StoreBlob(blob cpi.BlobAccess, artType, hint string, global cpi.AccessSpec, ctx cpi.StorageContext) (cpi.AccessSpec, error) {
+	if b.spec == nil {
+		return nil, nil
+	}
+
+	mimeType := blob.MimeType()
+	if mime.MIME_TGZ != mimeType && mime.MIME_TGZ_ALT != mimeType {
+		return nil, nil
+	}
+
+	blobReader, err := blob.Reader()
+	if err != nil {
+		return nil, err
+	}
+	defer blobReader.Close()
+
+	data, err := io.ReadAll(blobReader)
+	if err != nil {
+		return nil, err
+	}
+
+	// read package.json from tarball to get name, version, etc.
+	log := logging.Context().Logger()
+	log.Debug("reading package.json from tarball")
+	var pkg *Package
+	pkg, err = prepare(data)
+	if err != nil {
+		return nil, err
+	}
+	tbName := pkg.Name + "-" + pkg.Version + ".tgz"
+	pkg.Dist.Tarball = b.spec.Url + pkg.Name + "/-/" + tbName
+	log.Debug("identified package: %s@%s", pkg.Name, pkg.Version)
```

Comment:
Use key/value pairs in arguments.
If you want a formatted message use {{<key>}} to refer to the value of a dedicated key.